evalml/pipelines/components/estimators/regressors/lightgbm_regressor.py

```diff
@@ -170,8 +169,9 @@ def fit(self, X, y=None):
         X_encoded = self._encode_categories(X, fit=True)
         if y is not None:
             y = infer_feature_types(y)
-        X_encoded = downcast_int_nullable_to_double(X_encoded)
-        self._component_obj.fit(X_encoded, y)
+        # --> dont love calling this after we encode categories - try putting it first?
+        X_d, y_d = self._handle_nullable_types(X_encoded, y)
```

Comment:
`_encode_categories` calls `infer_feature_types` for X the lightgbm estimators, and also has logic related to categorical columns that shouldn't apply to any BooleanNullable columns we convert to Categorical, so I think it makes sense to have the handle call occur after `_encode_categories`, but ideally we'd have the handle call before doing any other work.